### PR TITLE
Convert Hive AcidUtils.getAcidState to Trino

### LINKED ISSUE #7603: Standard code path in BackgroundHiveSplitLoader should not reference AcidInfo.Builder
Currently we are calling out to `AcidInfo.Builder.build()` method also for code path in `BackgroundHiveSplitLoader.loadPartition` when we are not reading from ORC ACID table. 
This should be changed to make the code flow in the method more obvious.

### PR DESCRIPTION
**Pull request by electrum**
## Related

Fixes #7603

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Review by dain — APPROVED** [review]
Minor comments, but looks good otherwise

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/AcidTables.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public static int readAcidVersionFile(FileSystem fs, Path deltaOrBaseDir)
+            throws IOException
+    {
+        Path formatFile = versionFilePath(deltaOrBaseDir);
+        if (!fs.exists(formatFile)) {
+            return 0;
+        }
+        try (var in = fs.open(formatFile)) {
+            byte[] bytes = new byte[1];
+            if (in.read(bytes) == 1) {
```
[the remark this answers is not in the document]
Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/AcidTables.java:103]: This is a copy of the Hive code. I'll change to `readFully` and note that Hive only reads a single byte.

File: lib/trino-filesystem/src/main/java/io/trino/filesystem/FileEntry.java [hunk 2]
```diff
@@ -13,14 +13,27 @@
...
 import static java.util.Objects.requireNonNull;
 
-public record FileEntry(String path, long length, long lastModified)
+public record FileEntry(String path, long length, long lastModified, List<BlockLocation> blockLocations)
 {
     public FileEntry
     {
         checkArgument(length >= 0, "length is negative");
         requireNonNull(path, "path is null");
+        blockLocations = ImmutableList.copyOf(requireNonNull(blockLocations, "blockLocations is null"));
```
[the remark this answers is not in the document]
Comment (electrum) [reply lib/trino-filesystem/src/main/java/io/trino/filesystem/FileEntry.java:29]: The Hive split generation assumes they exist and are covering. I could add some basic validation here.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/FileSystemTesting.java [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+import static java.lang.Math.toIntExact;
+
+// copied from org.apache.hadoop.hive.ql.io.orc.TestInputOutputFormat
+public class FileSystemTesting
+{
+    public static class MockBlock
+    {
+        int offset;
+        int length;
+        final String[] hosts;
```
[the remark this answers is not in the document]
Comment (electrum) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/FileSystemTesting.java:52]: I cleaned up all the warnings

**Pull request merged**